google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,260 @@
+            CustomerClient customerClient = googleAdsRow.getCustomerClient();
+            // The customer client making the request will be returned in the
+            // search result with a level of 0.
+            if (customerClient.getLevel().getValue() == 0) {
+              // Store the root customer client, which is the first encountered customer client
+              // that has level of 0.
+              if (rootCustomerClient == null) {
+                rootCustomerClient = customerClient;
+              }
+              // The steps below map parent and children accounts. Continue here so that managers
+              // accounts exclude themselves from the list of their children accounts.
+              continue;
+            }
+
+            // For all level-1 (direct child) accounts that are manager accounts, the above
+            // query will be run against them to create a map of managers to their
+            // child accounts for printing the hierarchy afterwards.
+            List<CustomerClient> customerClientChildren = new ArrayList<>();
+            // Updates the customerClientChildren with the list of children accounts if there
+            // exists an entry in the map for the given manager.
+            if (customerIdsToChildAccounts.containsKey(customerId)) {
+              customerClientChildren = customerIdsToChildAccounts.get(customerId);
+            }
+            // Adds the current child account to the list of the given manager's child accounts,
+            // which will be empty if the manager has not yet been added to the map.
+            customerClientChildren.add(customerClient);
+            // Inserts or updates the map for the given manager with the new list of child accounts.
+            customerIdsToChildAccounts.put(customerId, customerClientChildren);
+            // Checks is the child account is a manager itself so that it can later be processed
+            // and added to the map if it hasn't been already.
+            if (customerClient.getManager().getValue()) {
+              // A customer can be managed by multiple managers, so to prevent visiting the same
+              // customer multiple times, we need to check if it's already in the map.
+              boolean alreadyVisited =
+                  customerIdsToChildAccounts.containsKey(customerClient.getId().getValue());
+              if (!alreadyVisited && customerClient.getLevel().getValue() == 1) {
+                unprocessedCustomerIds.add(customerClient.getId().getValue());
+              }
+            }
+          }
+        }
+
+        if (!(rootCustomerClient == null)) {
+          System.out.printf(
+              "The hierarchy of customer ID %d is printed below.%n",
+              rootCustomerClient.getId().getValue());
+          printAccountHierarchy(rootCustomerClient, customerIdsToChildAccounts, depth);
+        } else {
+          System.out.println(
```

Comment:
I thought this was fixed in [b/141760006](https://b/141760006)? If this is still happening it's a bug that should be reported and fixed.